### PR TITLE
compiler.h: Add _ between format|printf|syslog|scanf|strftime and like

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
Signed-off-by: Xiang Xiao <xiaoxiang@xiaomi.com>

## Summary
align with other macro naming style, need merge with https://github.com/apache/nuttx/pull/7915

## Impact
format specifiler

### CODE REVIEW

File: system/spi/spitool.h [hunk 1]
```diff
@@ -177,7 +177,7 @@ ssize_t spitool_exchange(FAR struct spitool_s *spitool,
                          FAR const void *outbuffer, size_t noutbytes,
                          FAR void *inbuffer, size_t ninbytes);
 int spitool_printf(FAR struct spitool_s *spitool, const char *fmt, ...)
```
Comment (pkarashchenko) [inline system/spi/spitool.h:179]:
```suggestion
int spitool_printf(FAR struct spitool_s *spitool, FAR const char *fmt, ...)
```

File: system/i2c/i2ctool.h [hunk 2]
```diff
@@ -171,7 +171,7 @@ extern const char g_i2cxfrerror[];
 ssize_t i2ctool_write(FAR struct i2ctool_s *i2ctool, FAR const void *buffer,
                       size_t nbytes);
 int i2ctool_printf(FAR struct i2ctool_s *i2ctool, const char *fmt, ...)
```
Comment (pkarashchenko) [inline system/i2c/i2ctool.h:173]:
```suggestion
int i2ctool_printf(FAR struct i2ctool_s *i2ctool, FAR const char *fmt, ...)
```

**Review by pkarashchenko — APPROVED** [review]

File: system/spi/spitool.h (hunk 1 above)
Comment (xiaoxiang781216) [reply system/spi/spitool.h]: Done.

File: system/i2c/i2ctool.h (hunk 2 above)
Comment (xiaoxiang781216) [reply system/i2c/i2ctool.h]: Done.

File: interpreters/quickjs/qjsmini.c [hunk 3]
```diff
@@ -168,14 +168,14 @@ static void *js_trace_malloc(JSMallocState *s, size_t size)
 
   assert(size != 0);
 
-  if (unlikely(s->malloc_size + size > s->malloc_limit))
+  if (unlikely(s->alloc_size + size > s->malloc_limit))
```
Comment (pkarashchenko) [inline interpreters/quickjs/qjsmini.c:171]: how this change is related to this PR?

**Review by pkarashchenko — APPROVED** [review]

File: interpreters/quickjs/qjsmini.c (hunk 3 above)
Comment (xiaoxiang781216) [reply interpreters/quickjs/qjsmini.c]: My fault, revert the change.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**